pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1584,15 +1585,48 @@ def _check_consider_using_with(self, node: nodes.Call) -> None:
         if could_be_used_in_with and not _will_be_released_automatically(node):
             self.add_message("consider-using-with", node=node)
 
-    def _check_use_list_or_dict_literal(self, node: nodes.Call) -> None:
-        """Check if empty list or dict is created by using the literal [] or {}."""
-        if node.as_string() in {"list()", "dict()"}:
+    def _check_use_list_literal(self, node: nodes.Call) -> None:
+        """Check if empty list is created by using the literal []."""
+        if node.as_string() == "list()":
             inferred = utils.safe_infer(node.func)
             if isinstance(inferred, nodes.ClassDef) and not node.args:
                 if inferred.qname() == "builtins.list":
                     self.add_message("use-list-literal", node=node)
-                elif inferred.qname() == "builtins.dict" and not node.keywords:
-                    self.add_message("use-dict-literal", node=node)
+
+    def _check_use_dict_literal(self, node: nodes.Call) -> None:
+        """Check if dict is created by using the literal {}."""
+        if not isinstance(node.func, astroid.Name):
+            return
+        inferred = utils.safe_infer(node.func)
+        if (
+            isinstance(inferred, nodes.ClassDef)
+            and inferred.qname() == "builtins.dict"
+            and not node.args
+        ):
+            self.add_message(
+                "use-dict-literal",
+                args=(self._dict_literal_suggestion(node),),
+                node=node,
+                confidence=INFERENCE,
+            )
+
+    @staticmethod
+    def _dict_literal_suggestion(node: nodes.Call) -> str:
+        elements = []
+        for keyword in node.keywords:
+            if keyword not in node.kwargs:
+                elements.append(f'"{keyword.arg}": {keyword.value.as_string()}')
+            if len(", ".join(elements)) >= 64:
+                break
+        for keyword in node.kwargs:
+            elements.append(f"**{keyword.value.as_string()}")
+            if len(", ".join(elements)) >= 64:
+                break
+        suggestion = f'{{{", ".join(elements)}}}'
+        if len(suggestion) > 64:
+            suggestion = f"{suggestion[:30]}…{suggestion[-30:]}"
```

Comment:
As the suggestion is close to 64 already (because we cut short previously after the one element that would make it more than 64) this might be better:
```suggestion
            suggestion = f"{suggestion}, … "
```
We won't cut any key/value this way and it's ~= 64. Let me know what you think

Reply:
Yes, makes sense. There's gonna be odd cases one way or the other.

E.g.: a dict with only one key/value pair that has combined 500 characters or something. Those will look silly, no matter what we do.